### PR TITLE
replaced user command with kubeconfig

### PR DESCRIPTION
**Pull request by jasmingacic**
Signed-off-by: jasmingacic <jasmin.gacic@gmail.com>
**What this PR Includes**
Replaces `k0s user create` with `k0s kubeconfig create` 

```
$ k0s kubeconfig 
Usage:
  k0s kubeconfig create [username] [flags]

Examples:
        Command to create a kubeconfig for a user:
        CLI argument:
        $ k0s kubeconfig create [username]

        optionally add groups:
        $ k0s kubeconfig create [username] --groups [groups]

Flags:
      --groups string   Specify groups

Global Flags:
  -c, --config string            config file (default: ./k0s.yaml)
      --data-dir string          Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
  -d, --debug                    Debug logging (default: false)
  -l, --logging stringToString   Logging Levels for the different components (default [kube-controller-manager=1,kube-scheduler=1,kubelet=1,etcd=info,containerd=info,konnectivity-server=1,kube-apiserver=1])
```
```
$ k0s kubeconfig create -h 
Create a kubeconfig with a signed certificate and public key for a given user (and optionally user groups)
Note: A certificate once signed cannot be revoked for a particular user

Usage:
  k0s kubeconfig create [username] [flags]

Examples:
        Command to create a kubeconfig for a user:
        CLI argument:
        $ k0s kubeconfig create [username]

        optionally add groups:
        $ k0s kubeconfig create [username] --groups [groups]

Flags:
      --groups string   Specify groups
  -h, --help            help for create

Global Flags:
  -c, --config string            config file (default: ./k0s.yaml)
      --data-dir string          Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
  -d, --debug                    Debug logging (default: false)
  -l, --logging stringToString   Logging Levels for the different components (default [konnectivity-server=1,kube-apiserver=1,kube-controller-manager=1,kube-scheduler=1,kubelet=1,etcd=info,containerd=info])
```

### CODE REVIEW

**Comment by jasmingacic** [comment]
I also removed `###### Auto generated by spf13/cobra on...`

**Comment by jasmingacic** [comment]
Added 
```
$ k0s kubeconfig admin
/var/lib/k0s/pki/admin.conf
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: ...
```

**Comment by jasmingacic** [comment]
```
$ k0s kubeconfig admin -h
Command dumps admin kubeconfig.

Usage:
  k0s kubeconfig admin [command] [flags]

Examples:
        $ k0s kubeconfig admin > kubeconfig
        $ export KUBECONFIG=kubeconfig
        $ kubectl get nodes

Flags:
  -h, --help   help for admin

Global Flags:
  -c, --config string            config file (default: ./k0s.yaml)
      --data-dir string          Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
  -d, --debug                    Debug logging (default: false)
  -l, --logging stringToString   Logging Levels for the different components (default [kube-apiserver=1,kube-controller-manager=1,kube-scheduler=1,kubelet=1,etcd=info,containerd=info,konnectivity-server=1])
```

File: cmd/kubeconfig.go [hunk 1]
```diff
@@ -47,27 +49,27 @@ users:
     client-key-data: {{.ClientKey}}
 `))
 
-	// userCmd creates new certs and kubeConfig for a user
-	userCmd = &cobra.Command{
-		Use:   "user [command]",
+	// kubeconfigCmd creates new certs and kubeConfig for a user
+	kubeconfigCmd = &cobra.Command{
+		Use:   "kubeconfig [command]",
 		Short: "Manage user access",
```
Comment (trawler) [inline cmd/kubeconfig.go:55]: change to a more meaningful description, now that we are no longer using the term `user`

**Review by trawler — CHANGES_REQUESTED** [review]
When creating a user's kubeconfig with a custom data-dir, ran into a bug:
```
ubuntu@controller-0:~/k0s (cmd_kubeconfig) [ 7d04b98 ] $ ./k0s kubeconfig create testUser --data-dir=~/test
Error: failed to read cluster ca certificate, is the control plane initialized on this node?: open ~/test/pki/ca.crt: no such file or directo
```
But the file exists and can be read:
```
ubuntu@controller-0:~/k0s (cmd_kubeconfig) [ 7d04b98 ] $ ls -latrh ~/test/pki/ca.crt
-rw-r--r-- 1 ubuntu ubuntu 1.1K Nov 26 10:48 /home/ubuntu/test/pki/ca.crt
```

File: cmd/kubeconfig.go (hunk 1 above)
Comment (jasmingacic) [reply cmd/kubeconfig.go:55]: Maybe "Dump kubeconfig for desired user".
I'm having difficulty phrasing it correctly.

**Comment by trawler** [comment]
> When creating a user's kubeconfig with a custom data-dir, ran into a bug:
> 
> ```
> ubuntu@controller-0:~/k0s (cmd_kubeconfig) [ 7d04b98 ] $ ./k0s kubeconfig create testUser --data-dir=~/test
> Error: failed to read cluster ca certificate, is the control plane initialized on this node?: open ~/test/pki/ca.crt: no such file or directo
> ```
> 
> But the file exists and can be read:
> 
> ```
> ubuntu@controller-0:~/k0s (cmd_kubeconfig) [ 7d04b98 ] $ ls -latrh ~/test/pki/ca.crt
> -rw-r--r-- 1 ubuntu ubuntu 1.1K Nov 26 10:48 /home/ubuntu/test/pki/ca.crt
> ```

Interestingly, the command works when using the full path (without tilde) `--data-dir=/home/ubuntu/test`
Let's fix path parsing too :)

File: cmd/kubeconfig.go (hunk 1 above)
Comment (trawler) [reply cmd/kubeconfig.go:55]: maybe instead of "dump" use: `Print kubeconfig for desired user to stdout`

**Review by trawler — APPROVED** [review]
fixed some phrasing + rebased.

**Pull request merged**